### PR TITLE
Refactor ResposiveText

### PR DESCRIPTION
**Pull request by mani-w3w**
Remove ResponsiveText.kt
Added AutoSizeText
Changed What3wordsAddress to have Secondary Text and Content Changed What3wordsAddressListItem to use What3wordsAddress with secondary content

### CODE REVIEW

**Review by kaytran2992 — APPROVED** [review]
LGTM. Thanks mate

File: design-library/src/main/java/com/what3words/design/library/ui/components/What3wordsAddress.kt [hunk 1]
```diff
@@ -70,68 +89,130 @@ object What3wordsAddressDefaults {
...
+ * @param secondaryText The secondary text string to be displayed.
+ * @param slashesMargin A lambda function for setting the margin of slashes.
+ */
+@Composable
+fun What3wordsAddress(
+    words: String,
+    modifier: Modifier = Modifier,
+    colors: What3wordsAddressDefaults.Colors = What3wordsAddressDefaults.defaultColors(),
+    textStyles: What3wordsAddressDefaults.TextStyles = What3wordsAddressDefaults.defaultTextStyles(),
+    secondaryText: String,
```
Comment (kaytran2992) [inline design-library/src/main/java/com/what3words/design/library/ui/components/What3wordsAddress.kt:123]: Hi @mani-w3w, Should we expose the TexStyle of the secondaryText that helps us scale up the font size if need

Comment (mani-w3w) [reply design-library/src/main/java/com/what3words/design/library/ui/components/What3wordsAddress.kt:123]: Good point I will add that 👍

Comment (mani-w3w) [reply design-library/src/main/java/com/what3words/design/library/ui/components/What3wordsAddress.kt:133]: done @kaytran2992

Comment (kaytran2992) [reply design-library/src/main/java/com/what3words/design/library/ui/components/What3wordsAddress.kt:133]: LTGM

**Review by phpduy — CHANGES_REQUESTED** [review]
Should we check the CircleCI tasks are failing mate?

**Review by phpduy — APPROVED** [review]
LGTM 👍

**Review by prosper-w3w — APPROVED** [review]

**Review by prosper-w3w — APPROVED** [review]
LGTM

**Pull request merged**